client/call.go

```diff
@@ -29,6 +30,7 @@ type Call struct {
 	Args  proto.Request  // The argument to the command
 	Reply proto.Response // The reply from the command
 	Err   error          // Error during call creation
+	post  func() error   // Function to be called after successful completion
```

Comment:
Yeah, this could be useful for general-purpose unmarshaling. I err on the side of not exporting until it is necessary.

As you noticed, passing the reply or call doesn't seem useful as you still need the closure for the thing you're unmarshaling into.